### ISSUE TITLE
Azure infra stack: single-region secure connectivity: AKS & AzureSQL, AzureCache, Buckets

### ISSUE THREAD

**Issue by negz** [crossplaneio/crossplane#617]
Azure infra stack users can configure everything for their environment from the Kubernetes API including networking, subnets, and secure connectivity between app deployments in a target cluster and the dynamically provisioned cloud services they depend on.

Part of the [0.3 release](https://github.com/crossplaneio/crossplane/blob/master/ROADMAP.md#v03---enable-partners-to-build-infra-stacks)

### Why? What problem are we solving?

A [Stack](https://github.com/crossplaneio/crossplane/blob/60333b/design/design-doc-stacks.md) is a set of CRDs and controllers that extend Crossplane. An infrastructure stack adds support for infrastructure (e.g. databases, VMs, etc), as opposed to adding support for an application (e.g. Wordpress, GitLab).

We'd like to take our first steps in this direction by breaking Crossplane's current Microsoft Azure managed resource functionality out into a Azure stack (covered by #612). Once the stack is broken out, we want to ensure it:

* Could be used by an application stack (such as a Wordpress stack) to deploy to Azure.
* Establishes a solid foundation for the community to build new Azure related controllers.

### What does it look like when we're done?

* The Azure stack contains `v1alpha2` variants of all existing `v1alpha1` Azure managed resources.
* The Azure stack contains `v1alpha2` variants of all existing `v1alpha1` Azure managed resource classes. This is predicated on the work tracked in #613.
* The Azure stack is based on the latest Crossplane best practices:
  * The stack is built on kubebuilder 0.2.0 (possibly a beta thereof), including controller-tools, and the controller-runtime.
  * All managed resources are implemented using the nascent (#603) generic managed resource reconciler.
  * All resource claims are implemented using the generic resource claim reconciler.
* An application owner can leverage the Azure infrastructure stack when deploying an app stack. e.g. A `KubernetesApplication` can be scheduled and deployed to an Azure backed `KubernetesCluster`, and leverage Azure backed resource claims.

Note that adding network connectivity support (per #606) is a stretch goal for this epic, unlike its AWS and GCP equivalents.

### How could this be demonstrated?

1. Crossplane user reads the blog post describing this new functionality
1. Click through to the updated user guide
1. Try out the updated example(s)

For example:
1. Begin with 'core' Crossplane, i.e. just the extension manager without any Azure support.
1. Install the Azure infrastructure stack.
1. Create, outside of Crossplane, any external resources necessary to ensure network connectivity (if applicable).
1. Create a `MySQLServerClass` that will use the aforementioned connectivity resources.
1. Create an `AKSClusterClass` that will use the aforementioned connectivity resources.
1. Create `MySQLInstance` and `KubernetesCluster` resource claims referencing the aforementioned resource classes.
1. Deploy wordpress to the aforementioned `KubernetesCluster`. Ensure it can access the aforementioned `MySQLInstance`.

### Related
* Assumes #613 handles: Create strongly typed resource classes for each managed kind
* [#642] Azure: Author security/connectivity [design doc](https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-resource-connectivity-mvp.md) with [Azure specifics](https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-resource-connectivity-mvp.md#microsoft-azure)
* [#647] Azure: Define new networking types and controllers
  * Virtual Network CRD and controller with [high fidelity](https://godoc.org/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-06-01/network#VirtualNetwork)
  * Subnet CRD and controller with [high fidelity](https://godoc.org/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-06-01/network#Subnet)
    * Must have ability to enable [Service Endpoints](https://docs.microsoft.com/en-us/azure/mysql/howto-manage-vnet-using-portal?toc=%2fazure%2fvirtual-network%2ftoc.json)
  * Virtual Network Rule CRD and controller with [high fidelity](https://godoc.org/github.com/Azure/azure-sdk-for-go/services/mysql/mgmt/2017-12-01/mysql#VirtualNetworkRule)
* [#648] Azure: High fidelity API and generic managed reconciler for all existing Azure resources (for v1alpha2)
  * Depends on #269 Converge on a consistent controller design
  * Version bump of Azure CRDs -> v1alpha2
* [#649] Azure: Validate e2e portable wordpress stack [#572 ] deployed securely & accessible
  * Connected Wordpress demo e2e
* [#650] Azure: Update examples and README files containing Wordpress examples.
  * configure classes, providers, default class policy, etc. 
  * use the portable #572 Wordpress example, that depends on default resource classes (omit classReference)
* [#656] Azure: Update API status table for infra stack capabilities, API version level, etc.
* Short blog post to blog.crossplane.io about this new functionality! #TBD
